src/js/components/Select/SelectContainer.js

```diff
@@ -255,19 +255,26 @@ class SelectContainer extends Component {
   };
 
   optionLabel = index => {
-    const { options, labelKey } = this.props;
+    const { options, labelKey, theme } = this.props;
     const option = options[index];
     let optionLabel;
+    let wrapInText = true;
     if (labelKey) {
       if (typeof labelKey === 'function') {
+        // we don't want to wrap in text if the caller is controlling the render
+        wrapInText = false;
```

Comment:
I'm wondering though, if the function might be doing something like translating a string into another string. I'm also thinking that technically, the option data might contain a React element. So, instead of basing the test on whether `labelKey` is a function, I'm thinking we should instead check on the type of `optionLabel`. If it is a react element, don't wrap it. Otherwise, if it is a string or number, wrap it in Text.